bridge-client/src/commonTest/kotlin/org/sagebionetworks/bridge/kmm/shared/repo/AdherenceRecordRepoTest.kt

```diff
@@ -44,23 +56,61 @@ class AdherenceRecordRepoTest: BaseTest() {
         runTest {
             val studyId = "testId"
             val scope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
-            val repo = AdherenceRecordRepo(getTestClient(json), null, ResourceDatabaseHelper(testDatabaseDriver()), scope)
+            val mockEngine = MockEngine.config {
+                // 1 - Load adherence call
+                addHandler (
+                    getJsonReponseHandler(json)
+                )
+                // 2 - Process updates call needs to fail so we still have pending local changes
+                addHandler { respondError(HttpStatusCode.GatewayTimeout) }
+                // 3 - Second Load adherence call
+                addHandler (
+                    getJsonReponseHandler(json)
+                )
+                reuseHandlers = false
+            }
+            val testClient = getTestClient(mockEngine)
+
+            val repo = AdherenceRecordRepo(testClient, null, ResourceDatabaseHelper(testDatabaseDriver()), scope)
 
             assertTrue(repo.loadRemoteAdherenceRecords(studyId))
-            val adherenceRecord = repo.getCachedAdherenceRecord(instanceGuid, startedOn)
+            var adherenceRecord = repo.getCachedAdherenceRecord(instanceGuid, startedOn)
             assertNotNull(adherenceRecord)
 
-            val recordMap =  repo.getCachedAdherenceRecords(listOf(instanceGuid), studyId)
+            val recordMap =  repo.getAllCachedAdherenceRecords(studyId).first()
             assertEquals(1, recordMap.size)
-            val recordList = recordMap.get(instanceGuid)
+            val recordList = recordMap[instanceGuid]
             assertNotNull(recordList)
-            val record = recordList.get(0)
+            val record = recordList[0]
             assertNotNull(record)
             assertEquals(instanceGuid, record.instanceGuid)
+
+            val updatedRecord = record.copy(clientData = JsonPrimitive("Test data"))
```

Comment:
Can you change this to set the clientData as a JsonObject rather than a primitive? Since we need to be able to merge scheduling data that we set when updating the start times with app data that the app uses, this is always going to be an object. Thanks.